src/dag-node/rmLink.js

```diff
@@ -4,12 +4,26 @@ const CID = require('cids')
 const { Buffer } = require('buffer')
 
 const rmLink = (dagNode, nameOrCid) => {
+  let predicate = null
+
   // It's a name
   if (typeof nameOrCid === 'string') {
-    dagNode._links = dagNode._links.filter((link) => link.Name !== nameOrCid)
+    predicate = link => link.Name === nameOrCid
```

Comment:
minor nit: the style in this library is to always-paren arrow functions, it'd be great to keep it that way (a rule I'd love enforced by standard tbh, for readability, but for now it's on a per-library convention basis). There's another instance 2 lines below too.